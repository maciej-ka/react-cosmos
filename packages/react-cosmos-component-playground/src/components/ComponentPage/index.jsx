import React, { Component } from 'react';
import { objectOf, arrayOf, string } from 'prop-types';
import LoaderGrid from '../LoaderGrid';

export default class ComponentPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gridData: this.getGridData(props)
    };
  }

  getGridData(props) {
    const {fixtures, component} = props;
    if (!fixtures[component]) {
      return [];
    }

    return fixtures[component].map(fixture => {
      return {
        fixture,
      };
    });
  }

  render() {
    return (
      <div>
        ComponentPage
        {JSON.stringify(this.state.gridData)}
        <LoaderGrid fixtures={this.state.gridData}/>
      </div>
    );
  }
}

ComponentPage.propTypes = {
  fixtures: objectOf(arrayOf(string))
};
