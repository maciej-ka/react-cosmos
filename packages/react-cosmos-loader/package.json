{
  "name": "react-cosmos-loader",
  "version": "2.0.0-beta.20",
  "description": "Loader for rendering React components in isolation",
  "repository": "https://github.com/react-cosmos/react-cosmos/tree/master/packages/react-cosmos-loader",
  "license": "MIT",
  "files": [
    "lib"
  ],
  "main": "lib/index.js",
  "dependencies": {
    "lodash.merge": "^4.6.0",
    "prop-types": "^15.5.10",
    "react-cosmos-state-proxy": "^2.0.0-beta.20",
    "react-cosmos-utils": "^2.0.0-beta.20"
  },
  "xo": false
}
