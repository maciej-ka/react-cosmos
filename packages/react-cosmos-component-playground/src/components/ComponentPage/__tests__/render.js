import React from 'react';
import { Loader } from 'react-cosmos-loader';
import renderer from 'react-test-renderer';
import ComponentPage from '../';
import blankFixture from '../__fixtures__/blank';

describe('ComponentPage blank', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(
      <Loader component={ComponentPage} fixture={blankFixture} />
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
