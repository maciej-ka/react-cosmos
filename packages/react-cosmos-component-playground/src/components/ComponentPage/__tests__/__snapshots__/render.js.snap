// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentPage blank renders correctly 1`] = `
<div>
  ComponentPage
  []
  <div>
    Grid of loaders
  </div>
</div>
`;
